### ISSUE TITLE
Changing (highmemory) scene cause GDscript to force to run next line and crash?

### ISSUE THREAD

**Issue by Qws** [godotengine/godot#9253]
**Operating system or device - Godot version:**
Windows 10. 
Godot 2.1.3
Pure Node2D

**Issue description:**
Code of the old scene keeps going even after I call this function: 

`get_tree().change_scene("res://Game/Credits.tscn")`

**Steps to reproduce:**

So I've SceneA and SceneB

SceneA use 1.5 GB ram when it's running (contains heavy assets made in DragonBones). So when I change from SceneA to SceneB with "get_tree().change_scene()", GDscript just continue the code of SceneA till it crashes, doesn't even have time to load SceneB.

When I test the same code again in HTML5 version it works... the scene does change, but according to the debugger the code in SceneA kept going till it crashed (just almost immediately). But at least it loads the SceneB and game is still functional.

This function I cast from SceneA:
```
func goodcredits():
	print("==============starting the credits===============")
	var nodesingroup = get_tree().get_nodes_in_group("WelcomeMap")
	nodesingroup.empty()
	nodesingroup.clear()
	get_tree().change_scene("res://Game/Credits.tscn")
	print("================================================")
```
Is this a bug? Or do I have to destroy/clear SceneA somehow before I change to SceneB?

**Comment by Zylann** [comment]
I had a problem too when I reached 1gb of RAM allocated in Godot: https://github.com/godotengine/godot/issues/8279
it just crashes beyond that point, which is too bad because computers have more memory than that these days^^"

**Comment by Qws** [comment]
Well, I started a new project, to test change scenes with nearly no MB usage. And yes, the code of the old scene didn't kept running when it changes to a new lightweight scene. So, problem doesn't exist in low usage of RAM...

Ehm... does Godot really get unstable when scenes starting to use 1GB+ RAM? 

To me it seems that "get_tree().change_scene("")" doesn't block the current thread but instead it starts new thread skipping "get_tree().change_scene()" code on when loading the next scene takes too long. Similar like Async code...

I guess I have to work around it... for now?

**Comment by kubecz3k** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about its current status here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
> Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

As there was no update since the previous post, we close this issue.

If it is still relevant in the way it was described and discussed above, please comment to ask for it to be reevaluated. If it is only partly relevant, or if the original issue has changed, it would be better to open a new issue (potentially referring to this one) focused on the current state.
